pkg/controller/job/job_controller.go

```diff
@@ -535,7 +541,14 @@ func (jm *Controller) syncJob(key string) (bool, error) {
 		jm.recorder.Event(&job, v1.EventTypeWarning, failureReason, failureMessage)
 	} else {
 		if jobNeedsSync && job.DeletionTimestamp == nil {
-			active, manageJobErr = jm.manageJob(activePods, succeeded, &job)
+			var pendingIndexes []bool
+			if job.Spec.Completions != nil && job.Spec.CompletionMode == batch.IndexedCompletion {
+				pendingIndexes, err = calculatePendingIndexes(pods, *job.Spec.Completions)
```

Comment:
call this inside manageJob? it doesn't see to be used in this function.

Reply:
Refactored